cinema/tests/test_actor_api.py

```diff
@@ -0,0 +1,105 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from cinema.models import Actor
+from cinema.serializers import ActorSerializer
+from user.tests.test_user_api import create_user
+
+ACTOR_URL = reverse("cinema:actor-list")
+
+
+def sample_actor(**params):
+    defaults = {
+        "first_name": "test_name",
+        "last_name": "test_last",
+    }
+    defaults.update(params)
+
+    return Actor.objects.create(**defaults)
+
+
+def detail_url(actor_id):
+    return reverse("cinema:actor-detail", args=[actor_id])
+
+
+class PublicActorApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(ACTOR_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class PrivateActorApiTests(TestCase):
+    def setUp(self):
+        self.user = create_user(
+            username="test_admin",
+            email="test@test.com",
+            password="testpass",
+        )
+        self.client = APIClient()
+        self.client.force_authenticate(user=self.user)
+
+    def test_get_actors(self):
+        sample_actor()
+
+        response = self.client.get(ACTOR_URL)
+
+        actors = Actor.objects.all()
+        serializer = ActorSerializer(actors, many=True)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_post_actors(self):
+        payload = {
+            "first_name": "test_name",
+            "last_name": "test_last",
+        }
+
+        response = self.client.post(ACTOR_URL, payload)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminActorApiTests(TestCase):
+    def setUp(self):
+        self.user = create_user(
+            username="test_admin",
+            email="test@test.com",
+            password="testpass",
+            is_staff=True,
+        )
+        self.client = APIClient()
+        self.client.force_authenticate(user=self.user)
+
+    def test_post_actors(self):
+        payload = {
+            "first_name": "test_name",
+            "last_name": "test_last",
+        }
+
+        response = self.client.post(ACTOR_URL, payload)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_retrieve_actor(self):
+        sample_actor()
+
+        response = self.client.get("http://127.0.0.1:8000/api/cinema/actors/1/")
```

Comment:
It is a bad practice - to specify full path in tests.
Imagine we push code to production, change prot, HTTP -> HTTPS.
And want to check, that tests are passing.
I'd rather do smth next:
```python
response = self.client.get(f"{ACTOR_URL}/1/")
```

Reply:
This is also a small cheat, but for this purpose is ok